mono/mini/m2n-gen.cs

```diff
@@ -174,7 +176,54 @@ class Driver {
 
 			Console.WriteLine ("\n}\n");
 		}
+		WriteDispatch (cookies);
+	}
+
+	static void WriteDispatch (string [] items)
+	{
+		Array.Sort (items);
+		Console.WriteLine ("static void\nicall_trampoline_dispatch (const char *cookie, void *target_func, InterpMethodArguments *margs)");
+		Console.WriteLine ("{");
+		Partition (items, 0);
+		Console.WriteLine ("\tprintf (\"CANNOT HANDLE COOKIE %s\\n\", cookie);");
+		Console.WriteLine ("\tg_assert (0);");
```

Comment:
This should use `g_error ("...")`. In general, `printf` doesn't work as expected on every platform so we use `g_print` instead. `g_error` will use it accordingly.